Deprecate cassandra module/returner

waynew opened the pull request:
### What does this PR do?

Deprecates the entirely broken cassandra module, directing users to cassandra_cql - the underlying library pycassa https://github.com/pycassa/pycassa has been deprecated for about 8 years now.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the [test documentation](https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite. -->
- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
~- [ ] Tests written/updated~ N/A

### Commits signed with GPG?
Yes

waynew committed:
Deprecate cassandra module/returner

waynew committed:
add changelog

waynew committed:
Update cassandra docs with deprecation warning

s0undt3ch commented on salt/modules/cassandra_mod.py:
```diff
@@ -27,6 +34,13 @@
     pass
 
 
+warn_until_date(
```
Won't this be issued everytime the loader tried to load it?
That might be a tad too noisy.

How about a decorator that triggers the warning if any of the functions are used?

s0undt3ch commented on salt/returners/cassandra_return.py:
```diff
@@ -42,6 +48,12 @@
+    "The cassandra returner is broken and deprecated, and will be removed"
+    " after {date}. Use the cassandra_cql returner instead",
+)
+
```
Same as above, unless for returners is way less noisy.

waynew replied:
I can put it in `__virtual__` for the success state? Or does that also get called every time the module is loaded?

s0undt3ch replied:
Virtual is also called everytime the module is loaded.
But, in this case I'm ok with the noise, because the module will only load if the dependency is available right?

waynew committed:
Move message to __virtual__

This should help cut down on any noise, specifically it will only warn
for folks that actually have the module installed.

waynew commented:
@s0undt3ch moved the warn in both cases to the __virtual__, after checking if the defunct pycassa python module is installed.
